kroxylicious-kms-provider-aws-kms/src/main/java/io/kroxylicious/kms/provider/aws/kms/AwsKmsKms.java

```diff
@@ -0,0 +1,226 @@
+
+    AwsKmsKms(URI awsUrl, String accessKey, String secretKey, String region, Duration timeout, SSLContext sslContext) {
+        Objects.requireNonNull(awsUrl);
+        Objects.requireNonNull(accessKey);
+        Objects.requireNonNull(secretKey);
+        Objects.requireNonNull(region);
+        this.awsUrl = awsUrl;
+        this.accessKey = accessKey;
+        this.secretKey = secretKey;
+        this.region = region;
+        this.timeout = timeout;
+        client = createClient(sslContext);
+    }
+
+    HttpClient createClient(SSLContext sslContext) {
+        HttpClient.Builder builder = HttpClient.newBuilder();
+        if (sslContext != null) {
+            builder.sslContext(sslContext);
+        }
+        return builder
+                .followRedirects(HttpClient.Redirect.NORMAL)
+                .connectTimeout(timeout)
+                .build();
+    }
+
+    /**
+     * {@inheritDoc}
+     * <br/>
+     * @see <a href="https://docs.aws.amazon.com/kms/latest/APIReference/API_GenerateDataKey.html">https://docs.aws.amazon.com/kms/latest/APIReference/API_GenerateDataKey.html</a>
+     */
+    @NonNull
+    @Override
+    public CompletionStage<DekPair<AwsKmsEdek>> generateDekPair(@NonNull String kekRef) {
+        final GenerateDataKeyRequest generateRequest = new GenerateDataKeyRequest(kekRef, "AES_256");
+        var request = createRequest(generateRequest, TRENT_SERVICE_GENERATE_DATA_KEY);
+        return sendAsync(kekRef, request, GENERATE_DATA_KEY_RESPONSE_TYPE_REF, UnknownKeyException::new)
+                .thenApply(response -> {
+                    var key = DestroyableRawSecretKey.takeOwnershipOf(response.plaintext(), AES_KEY_ALGO);
+                    return new DekPair<>(new AwsKmsEdek(kekRef, response.ciphertextBlob()), key);
+                });
+    }
+
+    /**
+     * {@inheritDoc}
+     * <br/>
+     * @see <a href="https://docs.aws.amazon.com/kms/latest/APIReference/API_Decrypt.html">https://docs.aws.amazon.com/kms/latest/APIReference/API_Decrypt.html</a>
+     */
+    @NonNull
+    @Override
+    public CompletionStage<SecretKey> decryptEdek(@NonNull AwsKmsEdek edek) {
+        final DecryptRequest decryptRequest = new DecryptRequest(edek.kekRef(), edek.edek());
+        var request = createRequest(decryptRequest, TRENT_SERVICE_DECRYPT);
+        return sendAsync(edek.kekRef(), request, DECRYPT_RESPONSE_TYPE_REF, UnknownKeyException::new)
+                .thenApply(response -> DestroyableRawSecretKey.takeOwnershipOf(response.plaintext(), AES_KEY_ALGO));
+    }
+
+    /**
+     * {@inheritDoc}
+     * <br/>
+     * @see <a href="https://docs.aws.amazon.com/kms/latest/APIReference/API_DescribeKey.html">https://docs.aws.amazon.com/kms/latest/APIReference/API_DescribeKey.html</a>
+     */
+    @NonNull
+    @Override
+    public CompletableFuture<String> resolveAlias(@NonNull String alias) {
+        final DescribeKeyRequest resolveRequest = new DescribeKeyRequest(ALIAS_PREFIX + alias);
```

Comment:
This currently is restricted to resolving AWS key alias that reside in the same customer account.  This is probably good enough for v1.

There's an argument for relaxing this and allowing the user to use any sort of key identifier.

* Alias
* Key Id
* ARN with Key Id
* ARN with Key Alias

https://docs.aws.amazon.com/kms/latest/APIReference/API_DescribeKey.html#API_DescribeKey_RequestSyntax

Allowing ARNs would be an enabler for cross AWS account use-cases. We'd need to also have the ability to serde ARN values. I've made a comment about this elsewhere.

Reply:
I don't think a serde should *ever* serialise an alias.  It should always be an immutable key identifier that gets serialised, or as close as we can get (https://github.com/kroxylicious/kroxylicious/issues/784).

In the case of AWS, I think ultimately we should be serialising the KeyId or an ARN with KeyId in a byte efficient pattern.  ARNs will let us support cross account use-cases.   I am not suggesting this for this PR, I think serialising the string form of the KeyId is good enough.